revscoring/dependencies/dependent.py

```diff
@@ -0,0 +1,36 @@
+import logging
+from functools import wraps
+
+logger = logging.getLogger("revscoring.dependent")
+
+def not_implemented(*args, **kwargs):
+    raise NotImplementedError("Not implemented.")
+
+class Dependent:
+
+    def __init__(self, name, process=not_implemented, depends_on=None,
+                             dependencies=None):
+        self.name = name
+        self.process = process
+        self.dependencies = dependencies or depends_on or []
+        self.calls = 0
+
+    def __call__(self, *args, **kwargs):
+        logger.debug("Executing {0}.".format(self))
+        self.calls += 1
+        return self.process(*args, **kwargs)
+
+    def __hash__(self):
+        return hash(('dependent', self.name))
+
+    def __eq__(self, other):
+        return hash(self) == hash(other)
+
+    def __ne__(self, other):
+        return not self == other
```

Comment:
It turns out that __ne__ is using hashing.  The == translates to a call to __eq__